File: 2260-divide-a-string-into-groups-of-size-k/divide-a-string-into-groups-of-size-k.java
class Solution {
    public String[] divideString(String s, int k, char fill) {
        int n=s.length(), len=(int)Math.ceil((double)n/k);
        String[] wd = new String[len];
        int l=0;
        for(int i=0;i<n;)
        {
            String w="";
            for(int j=0;j<k && i<n;j++,i++)
            {
                w+=s.charAt(i);
            }
            //System.out.println(w+" "+i+" "+l);
            wd[l]=w;
            l++;
        }
        while(wd[l-1].length()!=k)
        wd[l-1]+=fill;

        return wd;
    }
}

File: 2260-divide-a-string-into-groups-of-size-k/divide-a-string-into-groups-of-size-k.cpp
class Solution {
public:
    vector<string> divideString(string s, int k, char fill) {
        vector<string> wd;
        for(int i=0;i<s.size();)
        {
            string w="";
            for(int j=0;j<k && i<s.size();j++,i++)
            {
                 w+=s[i];
            }
            wd.push_back(w);
        }
        while(wd[wd.size()-1].size()!=k)
        {
            wd[wd.size()-1]+=fill;
        }
        return wd;
    }
};